check_stability.py

```diff
@@ -109,126 +106,8 @@ def __exit__(self, type, value, traceback):
-        if headers is None:
-            headers = {}
-        rv = self.headers.copy()
-        rv.update(headers)
-        return rv
-
-    def post(self, url, data, headers=None):
-        """Serialize and POST data to given URL."""
-        logger.debug("POST %s" % url)
-        if data is not None:
-            data = json.dumps(data)
-        resp = requests.post(
-            url,
-            data=data,
-            headers=self._headers(headers),
-            auth=self.auth
-        )
-        resp.raise_for_status()
-        return resp
-
-    def patch(self, url, data, headers=None):
-        """Serialize and PATCH data to given URL."""
-        logger.debug("PATCH %s" % url)
-        if data is not None:
-            data = json.dumps(data)
-        resp = requests.patch(
-            url,
-            data=data,
-            headers=self._headers(headers),
-            auth=self.auth
-        )
-        resp.raise_for_status()
-        return resp
-
-    def get(self, url, headers=None):
-        """Execute GET request for given URL."""
-        logger.debug("GET %s" % url)
-        resp = requests.get(
-            url,
-            headers=self._headers(headers),
-            auth=self.auth
-        )
-        resp.raise_for_status()
-        return resp
-
-    def post_comment(self, issue_number, body):
-        """Create or update comment in appropriate GitHub pull request comments."""
-        user = self.get(urljoin(self.base_url, "/user")).json()
-        issue_comments_url = urljoin(self.base_url, "issues/%s/comments" % issue_number)
-        comments = self.get(issue_comments_url).json()
-        title_line = format_comment_title(self.product)
-        data = {"body": body}
-        for comment in comments:
-            if (comment["user"]["login"] == user["login"] and
-                comment["body"].startswith(title_line)):
-                comment_url = urljoin(self.base_url, "issues/comments/%s" % comment["id"])
-                self.patch(comment_url, data)
-                break
-        else:
-            self.post(issue_comments_url, data)
-
-
-class GitHubCommentHandler(logging.Handler):
-
-    """GitHub pull request comment handler.
-
-    Subclasses logging.Handler to add ability to post comments to GitHub.
-    """
-
-    def __init__(self, github, pull_number):
-        """Extend logging.Handler and set required properties on self."""
-        logging.Handler.__init__(self)
-        self.github = github
-        self.pull_number = pull_number
-        self.log_data = []
-
-    def emit(self, record):
-        """Format record and add to log"""
-        try:
-            msg = self.format(record)
-            self.log_data.append(msg)
-        except Exception:
-            self.handleError(record)
-
-    def send(self):
-        """Post log to GitHub and flush log."""
-        self.github.post_comment(self.pull_number, "\n".join(self.log_data))
-        self.log_data = []
-
-
-class Browser(object):
```

Comment:
Can we leave this in but convert it to an [Abstract Base Class](https://docs.python.org/2/library/abc.html], and add whatever methods we expect to be implemented by all subclasses decorated with `abstractmethod`.